openssl/src/ssl/mod.rs

```diff
@@ -1968,6 +2025,22 @@ impl ForeignTypeRef for SslCipherRef {
 }
 
 impl SslCipherRef {
+    /// Returns an SslCipher matching the two-byte TLS cipher ID (allocated by the IANA, in network
+    /// byte order).
+    #[corresponds(SSL_CIPHER_find)]
+    pub fn find<'a>(ssl: &'a SslRef, cipher_id: &[u8]) -> Result<&'a SslCipherRef, ErrorStack> {
+        unsafe {
+            let ssl = ssl.as_ptr();
+            let ptr = ffi::SSL_CIPHER_find(ssl, cipher_id.as_ptr());
```

Comment:
Something needs to ensure that cipher_id is (at least) two bytes.

Reply:
My understanding of the documentation was that OpenSSL would set an error if the cipher ID was invalid for any reason, I'll run some tests to check that assumption.

Can you point me toward an example of the type of error I should return if the cipher ID has an invalid format?

Reply:
Because `SSL_CIPHER_find()` doesn't seem to set internal OpenSSL errors I switched the return type from a `Result<>` to `Option<>`. I also added a length check to prevent returning a cipher for a too-short or too-long ID.

0e0f76bc6be9cfdfa8dcbf3b64524f28c09266f5